Sync/SyncTelemetryUtils.swift

```diff
@@ -384,3 +384,72 @@ public struct SyncPing: SyncTelemetryPing {
         }
     }
 }
+
+public class GleanSyncOperationHelper {
+    public init () {}
+
+    public func start() {
+        GleanMetrics.Sync.syncUuid.generateAndSet()
+    }
+
+    public func end(_ result: SyncOperationResult) {
+        if let engineResults = result.engineResults.successValue {
+            engineResults.forEach { result in
+                let (name, status) = result
+                switch status {
+                case .completed(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .partial(let stats):
+                    self.submitPingPerEngine(name, stats)
+                case .notStarted(let reason):
+                    ()
+                    // failure_reason would be recorded here for the temp-<engine>-sync pings
+                }
+            }
+        } else if let failure = result.engineResults.failureValue {
+            // failure_reason would be recorded here for the temp-sync ping
+        }
+        
+        GleanMetrics.Pings.tempSync.submit()
+    }
+    
+    private func submitPingPerEngine(_ engineName: String, _ stats: SyncEngineStatsSession) {
+        // Create maps on labels to stat value,
+        // keeping only the values that are above zero.
+        //
+        // If we attempt to add 0 to a Glean counter,
+        // Glean will record an error. We don't want that here.
+        let incomingLabelsToValue = [
+            // TODO: What is the difference between applied and succeeded?
```

Comment:
@lougeniaC64 What is the difference between applied and succeeded?

I am adding both here, but I that is probably wrong.

Reply:
Both statuses are valid. [This](https://searchfox.org/mozilla-central/source/services/sync/modules/engines.js#1235) is an explanation of both `applied` and `newFailed`. In short, `applied` is the number of incoming records that should be added or to which existing records should be updated. `newFailed` represents the number of new failures. Which leaves `succeeded` and `failed` to represent the number of failures or success overall.

Reply:
Ok, I'll stick to `succeded` and `failed` only here, because I think that makes sense. Would you agree?